docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md

```diff
@@ -9,25 +9,245 @@ menu:
+
+- *yb_follower_read_staleness_ms* : This session variable defines the maximum staleness that the user is willing to tolerate.
+
+### Default behavior
+
+- *yb_read_from_followers* : Defaults to false.
+
+- *yb_follower_read_staleness_ms* : Defaults to 30 seconds.
+
+### Expected behavior
+
+The table describes what the expected behavior is when a read happens from a follower.
+
+| Conditions | Expected behavior |
+| --- | --- |
+| yb_read_from_followers is true AND <br> (Transaction is marked read-only <br>OR<br> Single-statement-select can be inferred to be read-only) <br> | Read will happen from the closest follower|
+|yb_read_from_followers is false OR <br> Transaction/statement is not read only | Read will happen from the leader |
+
+### Read from follower conditions
+
+- If the follower’s safe-time is larger than the read-time thus chosen, the follower may serve the read without any delay.
+
+- If the follower is not yet caught up to the specified timestamp, the read will be redirected to a different replica transparently from the end-user. The end user may see a slight increase in latency depending on the location of the replica which satisfies the read.
+  - The current behavior is that a follower/replica would wait for the requested timestamp to become safe. This behavior shall be changed to respond back to the `ybclient/pggate` immediately, so that the request may be routed to a different server that has the data.
+  - The current redirection policy is to directly go to the leader if the follower read is rejected by a follower. This behavior could be also be changed going forward, to explore other replicas before going to the leader.
+  - If the request reaches the leader, it will no longer reject/redirect the request.
+
+### Read-only transaction conditions
+
+Regarding marking the transaction as read only, a user can do one of the following:
+
+- `SET TRANSACTION READ ONLY` : Applies only to the current transaction block.
+- `SET SESSION CHARACTERISTICS AS TRANSACTION READ ONLY` : Applies the read-only setting for all statements and transaction blocks that follow.
+- `SET default_transaction_read_only = TRUE` : Applies the read-only setting for all statements and transaction blocks that follow.
+
+## Examples
+
+This example uses follower reads since the **transaction** is marked read-only.
+
+```sql
+set yb_read_from_followers = true;
+start transaction read only;
+SELECT * from t WHERE k='k1';
+commit;
+```
+
+```output
+ k  | v
+----+----
+ k1 | v1
+```
+
+This example uses follower reads since the **session** is marked read only.
+
+```sql
+set session characteristics as transaction read only;
+set yb_read_from_followers = true;
+SELECT * from t WHERE k='k1';
+```
+
+```output
+ k  | v
+----+----
+ k1 | v1
+(1 row)
+```
+
+This next two examples use follower reads since **pg_hint** can be used during PREPARE and CREATE FUNCTION to do follower reads.
```

Comment:
Can't pg_hint method be used for a top-level/standalone SELECT statement? Does it have to be only via PREPARE or CREATE FUNCTIONA?

Reply:
Yes. the pg_hint method can also be used for stand_alone statements.

The important point to highlight there is that for Prepare/Functions, the pg_hint needs to be applied at the prepare/function-definition stage. Not at the execute stage.

Reply:
I will include it as a note.